### PR TITLE
Implement ycodemirror binding using codeeditor-nbmodel

### PR DESCRIPTION
**Pull request by dmonad**

### CODE REVIEW

File: packages/codemirror/src/editor.ts [hunk 1]
```diff
@@ -7,6 +7,9 @@ import CodeMirror from 'codemirror';
 
 import { showDialog } from '@jupyterlab/apputils';
 
+import * as nbmodel from '@jupyterlab/nbmodel';
```
Comment (echarles) [inline packages/codemirror/src/editor.ts:10]: @dmonad As we have just discussed on a call, I write a reminder so you check if when modelDB is updated, that the y-codemirror content is also updated.

**Review by echarles — COMMENTED** [review]
As we discussed, it is fine for now to be specific with concrete yjs implementation class in the @jupyterlab/codemirror package. We can always revisit later on if we find it useful useful.

**Comment by echarles** [comment]
@dmonad I wonder if we could have a flag `useBinding` (set by default to true) that would tell to effectively use the CodeMirrorBinding. If the flag would be set to false, we would have the exact previous behavior/implementation. Looking at the changes of this PR, it sounds like this approach would be possible.

Later on, I can open a PR which would set the flag based on a user setting.

WDYT?

**Comment by dmonad** [comment]
Definitely possible because we are currently keeping full compatibility to `modeldb`. The disadvantage is that we would need to maintain multiple paths. There might be bugs that only happen with one of the bindings. So this will be more difficult to maintain in the future. Both provide the same functionality. But y-codemirror is able to apply multiple changes at once which is critical when syncing multiple changes.

We discussed that we want to eventually migrate from `modeldb` to `nbmodel`. Since there is no immediate upside of using the existing modeldb-based editor binding, I'd prefer to reduce complexity and remove it.

**Comment by echarles** [comment]
I was proposing to have more incremental steps so we can iterate and better understand what is going on. Of course, we don't want to maintain multiple path and we aim to simplify the code base. My favorite way would just be to have temporary that flag for one or two weeks, to explicitly and completely understand what it means to use CodeMirrorBinding. I am confident that we can remove that flag and the old code by those 1 or 2 weeks. Does not make sense then to plan a user setting for that, but as developer, I like to fully understand the changes. 

If it is not too difficult, I would favor have this PR with a flag set to true which enable by default the new behavior, still having the ability for a developer to set it to false to fallback to the old behavior. This would thus easily ensure we have a fallback to test any regression. It would also help in to ensure test are OK.

**Comment by echarles** [comment]
Just merged #3 which gives conflict now with this PR

**Comment by dmonad** [comment]
I kept the existing binding and also ignore modeldb events when the y-codemirror binding is used.

It is a bit awkward. But we can keep this flag for now. 

This PR is ready to be merged.

**Review by echarles — APPROVED** [review]
Thx for the USE_YCODEMIRROR_BINDING flag. I see it gone soon, but I love it for now as it clearly shows what it means to use yjs-codemirror.

**Pull request merged**